src/index.rs

```diff
@@ -199,6 +203,37 @@ impl Index {
 
 		self.load_module(module_name).await
 	}
+	/// Find all auto_install modules whose dependencies are satisfied by the given set of modules
+	fn find_auto_install_modules(&self, loaded_modules: &HashSet<ModuleName>) -> Vec<ModuleName> {
+		let mut auto_install_modules = Vec::new();
+
+		for root in self.roots.iter() {
+			for (&module_key, module_entry) in root.iter() {
+				if module_entry.auto_install && !loaded_modules.contains(&module_key) {
+					// Check if all dependencies are satisfied
+					let all_deps_satisfied = module_entry.dependencies.iter().all(|dep| loaded_modules.contains(dep));
```

Comment:
This is logically correct, but inhibits completions and gotodefs if not all modules are defined in dependencies list. Any suggestions for improving UX here?